src/Workspaces/Core/Portable/Workspace/Solution/FileTextLoader.cs

```diff
@@ -207,20 +200,25 @@ internal override TextAndVersion LoadTextAndVersionSynchronously(LoadTextOptions
 
             ValidateFileLength(Path, fileLength);
 
-            TextAndVersion textAndVersion;
+            var textAndVersion = FileUtilities.RethrowExceptionsAsIOException(
+                static t =>
+                {
+                    // Open file for reading with FileShare mode read/write/delete so that we do not lock this file.
+                    using var stream = t.self.OpenFile(bufferSize: 4096, useAsync: false);
+                    var version = VersionStamp.Create(t.prevLastWriteTime);
+                    var text = t.self.CreateText(stream, t.options, t.cancellationToken);
+                    return TextAndVersion.Create(text, version, t.self.Path);
+                }, (self: this, prevLastWriteTime, options, cancellationToken));
```

Comment:
same as above.  the sync path now wraps opening the file *and* reading it in the exception transformation helper.